### PR TITLE
Client

### PR DESCRIPTION
**Pull request by saladuit**
- state tracker in httpserver.cpp
- client state header
- to notify httpserver of change to be made to _poll_fd vector
- basic request parsing without body and content length nor chunking (moved and refactored @lucienvb  code to the class it belongs to
- simple "hello world" response class
- placed some comments where excpetion handling will be necessary
- files to pay attention to are HTTPServer, Client & HTTPRequest classes

### CODE REVIEW

File: src/HTTPRequest.cpp [hunk 1]
```diff
@@ -1,182 +1,126 @@
-#include "../include/HTTPRequest.hpp"
+#include "ClientState.hpp"
+#include <HTTPRequest.hpp>
 #include <Logger.hpp>
+#include <SystemException.hpp>
 #include <sstream>
-#include <string>
 
-HTTPRequest::HTTPRequest()
-	: _methodType(HTTPMethod::GET), _content_length(0), _content_length_cpy(0),
```
Comment (Tentanus) [inline src/HTTPRequest.cpp:7]: should be HTTPMethod::UNKOWN

**Review by Tentanus — CHANGES_REQUESTED** [review]
Great succes

**Review by saladuit — COMMENTED** [review]
Everythin was thus perfect

**Review by Tentanus — APPROVED** [review]
Great succes

**Pull request merged**